pkg/controller/elasticsearch/user/reconcile_test.go

```diff
@@ -114,7 +118,9 @@ func Test_aggregateFileRealm(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			c := k8s.NewFakeClient(sampleUserProvidedFileRealmSecrets...)
-			fileRealm, controllerUser, err := aggregateFileRealm(context.Background(), c, tt.es, initDynamicWatches(), record.NewFakeRecorder(10), testPasswordHasher, metadata.Metadata{})
+			generator, _ := password.NewGenerator(nil)
```

Comment:
See my suggestion above to code against an interface that allows you to use a simple implemenation in tests.